msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileSceneKitAssetsTaskBase.cs

```diff
@@ -195,6 +195,14 @@ public override bool Execute ()
 					// .. and remove the @OriginalItemSpec which is for @asset
 					scnassetsItem.RemoveMetadata ("OriginalItemSpec");
 
+					// The Link metadata is for the asset, but 'scnassetsItem' is the containing *.scnasset directory,
+					// so we need to update the Link metadata accordingly (if it exists).
+					var link = scnassetsItem.GetMetadata ("Link");
+					if (!string.IsNullOrEmpty (link)) {
+						link = link.Substring (0, link.Length - (asset.ItemSpec.Length - scnassets.Length));
```

Comment:
Will `link.Length` always be greater than `asset.ItemSpec.Length - scnassets.Length`?

Reply:
Fixed!